### ISSUE TITLE
Duplicate transaction

### ISSUE THREAD

**Issue by jorisa** [firefly-iii/firefly-iii#3562]
**Bug description**
I am running Firefly III version 5.3.0, and my problem is: I used the CSV importer to import 5 years worth of ING transactions. It timed out after a while so I split the CSV's into smaller files and restarted it, then I noticed a lot of duplicate transactions. Here is the raw json of an example duplicate:

https://pastebin.com/PeKaJctF

For a moment I thought it could be due to a category being assigned  between the two imports, but it looks like transactions that were not affected by rules  were also duplicated.

**Steps to reproduce**
- Use CSV importer
- Terminate import halfway
- Run automation to assign category
- Start again using same data

**Extra info**
Using freshly installed docker installation using docker-compose as described in the documentation.

**Bonus points**
I did compare the two jsons and too me they look identical so I dont understand why it was duplicated.

**Comment by JC5** [comment]
According to your JSON file, they *are* the same. They have the same hash value. Could you share your configuration file, and possibly some log entries? You can also send them to me.

When transactions are the same they're generally not imported, unless you disable this.

**Comment by jorisa** [comment]
I used this configuration:
https://github.com/firefly-iii/import-configurations/blob/main/nl/ing/default.json

Here is some log data, not sure what exactly you are looking for:
```
Now in entrypoint.sh for Firefly III
Entrypoint script version is 1.0.6 (2020-06-07)
Running as www-data.
Now parsing _FILE variables.
done!
Touch DB file (if SQLite)...
Dump auto load...
Generating optimized autoload files
Generated optimized autoload files containing 4911 classes
Discover packages...
Discovered Package: adldap2/adldap2-laravel
Discovered Package: davejamesmiller/laravel-breadcrumbs
Discovered Package: fideloper/proxy
Discovered Package: jc5/google2fa-laravel
Discovered Package: laravel/passport
Discovered Package: laravel/ui
Discovered Package: laravelcollective/html
Discovered Package: nesbot/carbon
Discovered Package: rcrowe/twigbridge
Package manifest generated successfully.
Wait for the database.
DB is up. Time to execute artisan commands.
Run various artisan commands...
Running migration commands...
Database "firefly" exists.
Migration table created successfully.
Migrating: 2016_06_16_000000_create_support_tables
Migrated:  2016_06_16_000000_create_support_tables (20.16 seconds)
Migrating: 2016_06_16_000001_create_users_table
Migrated:  2016_06_16_000001_create_users_table (0.14 seconds)
Migrating: 2016_06_16_000002_create_main_tables
Migrated:  2016_06_16_000002_create_main_tables (55.7 seconds)
Migrating: 2016_08_25_091522_changes_for_3101
Migrated:  2016_08_25_091522_changes_for_3101 (0.8 seconds)
Migrating: 2016_09_12_121359_fix_nullables
Migrated:  2016_09_12_121359_fix_nullables (0.45 seconds)
Migrating: 2016_10_09_150037_expand_transactions_table
Migrated:  2016_10_09_150037_expand_transactions_table (0.02 seconds)
Migrating: 2016_10_22_075804_changes_for_v410
Migrated:  2016_10_22_075804_changes_for_v410 (0.05 seconds)
Migrating: 2016_11_24_210552_changes_for_v420
Migrated:  2016_11_24_210552_changes_for_v420 (0.03 seconds)
Migrating: 2016_12_22_150431_changes_for_v430
Migrated:  2016_12_22_150431_changes_for_v430 (2.96 seconds)
Migrating: 2016_12_28_203205_changes_for_v431
Migrated:  2016_12_28_203205_changes_for_v431 (0.23 seconds)
Migrating: 2017_04_13_163623_changes_for_v440
Migrated:  2017_04_13_163623_changes_for_v440 (31.74 seconds)
Migrating: 2017_06_02_105232_changes_for_v450
Migrated:  2017_06_02_105232_changes_for_v450 (0.75 seconds)
Migrating: 2017_08_20_062014_changes_for_v470
Migrated:  2017_08_20_062014_changes_for_v470 (20.56 seconds)
Migrating: 2017_11_04_170844_changes_for_v470a
Migrated:  2017_11_04_170844_changes_for_v470a (0.65 seconds)
Migrating: 2018_01_01_000001_create_oauth_auth_codes_table
Migrated:  2018_01_01_000001_create_oauth_auth_codes_table (1.13 seconds)
Migrating: 2018_01_01_000002_create_oauth_access_tokens_table
Migrated:  2018_01_01_000002_create_oauth_access_tokens_table (4.58 seconds)
Migrating: 2018_01_01_000003_create_oauth_refresh_tokens_table
Migrated:  2018_01_01_000003_create_oauth_refresh_tokens_table (7.74 seconds)
Migrating: 2018_01_01_000004_create_oauth_clients_table
Migrated:  2018_01_01_000004_create_oauth_clients_table (0.12 seconds)
Migrating: 2018_01_01_000005_create_oauth_personal_access_clients_table
Migrated:  2018_01_01_000005_create_oauth_personal_access_clients_table (10.26 seconds)
Migrating: 2018_03_19_141348_changes_for_v472
Migrated:  2018_03_19_141348_changes_for_v472 (0.51 seconds)
Migrating: 2018_04_07_210913_changes_for_v473
Migrated:  2018_04_07_210913_changes_for_v473 (1.47 seconds)
Migrating: 2018_04_29_174524_changes_for_v474
Migrated:  2018_04_29_174524_changes_for_v474 (2.25 seconds)
Migrating: 2018_06_08_200526_changes_for_v475
Migrated:  2018_06_08_200526_changes_for_v475 (11.81 seconds)
Migrating: 2018_09_05_195147_changes_for_v477
Migrated:  2018_09_05_195147_changes_for_v477 (0.97 seconds)
Migrating: 2018_11_06_172532_changes_for_v479
Migrated:  2018_11_06_172532_changes_for_v479 (0.28 seconds)
Migrating: 2019_01_28_193833_changes_for_v4710
Migrated:  2019_01_28_193833_changes_for_v4710 (1.55 seconds)
Migrating: 2019_02_05_055516_changes_for_v4711
Migrated:  2019_02_05_055516_changes_for_v4711 (1.16 seconds)
Migrating: 2019_02_11_170529_changes_for_v4712
Migrated:  2019_02_11_170529_changes_for_v4712 (0.05 seconds)
Migrating: 2019_03_11_223700_fix_ldap_configuration
Migrated:  2019_03_11_223700_fix_ldap_configuration (0.03 seconds)
Migrating: 2019_03_22_183214_changes_for_v480
Migrated:  2019_03_22_183214_changes_for_v480 (1.92 seconds)
Migrating: 2019_11_30_000000_create_2fa_token_table
Migrated:  2019_11_30_000000_create_2fa_token_table (1.53 seconds)
Migrating: 2019_12_28_191351_make_locations_table
Migrated:  2019_12_28_191351_make_locations_table (0.06 seconds)
Migrating: 2020_03_13_201950_changes_for_v520
Migrated:  2020_03_13_201950_changes_for_v520 (32.17 seconds)
Migrating: 2020_06_07_063612_changes_for_v530
Migrated:  2020_06_07_063612_changes_for_v530 (8.55 seconds)
Migrating: 2020_06_30_202620_changes_for_v530a
Migrated:  2020_06_30_202620_changes_for_v530a (1.1 seconds)
Seeding: AccountTypeSeeder
Seeded:  AccountTypeSeeder (1.92 seconds)
Seeding: TransactionCurrencySeeder
Seeded:  TransactionCurrencySeeder (0.26 seconds)
Seeding: TransactionTypeSeeder
Seeded:  TransactionTypeSeeder (0.07 seconds)
Seeding: PermissionSeeder
Seeded:  PermissionSeeder (0.07 seconds)
Seeding: LinkTypeSeeder
Seeded:  LinkTypeSeeder (0.06 seconds)
Seeding: ConfigSeeder
[2020-07-13 22:54:52] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).  
Seeded:  ConfigSeeder (0.06 seconds)
Database seeding completed successfully.
Going to decrypt the database.
Decrypted the data in table "accounts".
Decrypted the data in table "attachments".
Decrypted the data in table "bills".
Decrypted the data in table "budgets".
Decrypted the data in table "categories".
Decrypted the data in table "piggy_banks".
Decrypted the data in table "preferences".
Decrypted the data in table "tags".
Decrypted the data in table "transaction_journals".
Decrypted the data in table "transactions".
Decrypted the data in table "journal_links".
Done!
Running upgrade commands...
All split journal transaction identifiers are correct.
Verified and fixed transaction identifiers in 0.66 seconds.
Found no split transaction journals. Nothing to do.
Migrate split journals to groups in 3.55 seconds.
No need to convert transaction journals.
Migrate all journals to groups in 0 seconds.
No journals to migrate to groups.
All accounts are OK.
Verified and fixed account currencies in 0.93 seconds.
All transfers have correct currency information.
Verified and fixed currency information for transfers in 2.66 seconds.
Verified 0 transaction(s) and journal(s).
Verified and fixed transaction currencies in 0.62 seconds.
No notes to migrate.
Migrated notes in 0.65 seconds.
All attachments are OK.
Migrated attachment notes in 1.59 seconds.
All bills are OK.
Verified and fixed bills in 0.46 seconds.
All budget limits are correct.
Verified budget limits in 0.09 seconds.
No incorrectly stored credit card liabilities.
Check 0 transaction journal(s) for budget info.
Check 0 transaction journal(s) for category info.
Updated category and budget info for all transaction journals in 0.09 seconds.
All account meta is OK.
Fixed account meta data in 0.3 seconds.
No recurrence meta data migrated.
Migrated recurrence meta data in 0.1 seconds.
Running verification commands...
All piggy bank events are correct.
Verified the content of 0 piggy bank events in 0.06 seconds.
All link types OK!
Verified link types in 0.07 seconds
All access tokens OK!
Verify access tokens in 0.04 seconds.
All transaction journals have correct bill information.
Verified bills / journals in 0.06 seconds
0 different currencies are currently in use.
All currencies are correctly enabled or disabled.
Verified currencies in 0.08 seconds.
No invalid budget/journal entries.
Verified budget/journals in 0.06 seconds.
Amount integrity OK!
Verified amount integrity in 0.03 seconds
No zero-amount transaction journals.
Verified zero-amount integrity in 0.06 seconds
No orphaned transactions.
No orphaned accounts.
Verified orphans in 0.06 seconds
No uneven transaction journals.
No empty transaction journals.
Verified empty journals in 0.01 seconds
Verified empty groups in 0.06 seconds
All account types are OK!
Verifying account types took 0.08 seconds
All meta fields are correct.
Renamed meta fields in 0.05 seconds
Going to verify 0 opening balance transactions.
There was nothing to fix in the opening balance transactions.
Verified all transaction group and journal title lengths in 0.06 seconds.
Corrected recurring transactions 0.07 seconds.
Running report commands...
Report on empty objects finished in 0.09 seconds
Report on total sum finished in 0.05 seconds
Generated and stored new keys.
Generating new OAuth keys...
Encryption keys already exist. Use the --force option to overwrite them.
Personal access client created successfully.
Client ID: 1
Client secret: SNIP
Password grant client created successfully.
Client ID: 2
Client secret: SNIP
Updated version.
Application cache cleared!
Configuration cache cleared!
Configuration cached successfully!
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v5.3.0!                                |
| This version of Firefly III requires PHP7.4.                                 |
|                                                                              |
+------------------------------------------------------------------------------+
Go!
```

**Comment by JC5** [comment]
What's the configuration you've used to import data using the CSV importer? Once you're done, it offers you to download a JSON file. Could you share that one?
